config/config.go

```diff
@@ -1,39 +1,279 @@
+		Password string `mapstructure:"password"`
+	}{
+		UserName: "dice",
+		Password: RequirePass,
+	},
+	Network: struct {
+		IOBufferLength    int `mapstructure:"iobufferlength"`
+		IOBufferLengthMAX int `mapstructure:"iobufferlengthmax"`
+	}{
+		IOBufferLength:    512,
+		IOBufferLengthMAX: 50 * 1024,
+	},
+}
 
-// Users and ACLs
-var (
-	// if RequirePass is set to an empty string, no authentication is required
-	RequirePass string = constants.EmptyStr
-)
+// DiceConfig is the global configuration object for dice
+var DiceConfig *Config
+
+func SetupConfig() {
+	if !VerifyConfigCmd() || !isConfigFilePresent() {
+		DiceConfig = &defaultConfig
+		log.Info("Using default configurations.")
+		return
+	}
+
+	setUpViperConfig(ConfigFileLocation)
+}
+
+func VerifyConfigCmd() bool {
+	if areBothFlagsSet() {
+		log.Error("Both -o and -c flags are set. Please use only one flag.")
+		return false
+	}
+
+	// if -o flag is set, create a config file at the specified path by verifying the path
+	if isValidDirPath() {
+		ConfigFileLocation = filepath.Join(CustomConfigFilePath, constants.DefaultConfigName)
+		createConfigFile(ConfigFileLocation)
+		return true
+	}
+
+	// if -c flag is set then viper will take care of reading the config file and verifying it
+	return true
+}
+
+func createConfigFile(configFilePath string) {
+	if _, err := os.Stat(configFilePath); err == nil {
+		log.Warnf("config file already exists at %s", configFilePath)
+		setUpViperConfig(configFilePath)
+		return
+	}
+
+	dir := filepath.Dir(configFilePath)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		log.Errorf("directory does not exist at %s", dir)
+		log.Warn("starting Dice with default configurations.")
+		DiceConfig = &defaultConfig
+		return
+	}
+
+	configFile, err := os.OpenFile(configFilePath, os.O_CREATE|os.O_RDWR, 0666)
+	if err != nil {
+		log.Errorf("failed to create config file at %s: %v", configFilePath, err)
+		log.Warn("starting Dice with default configurations.")
+		DiceConfig = &defaultConfig
+		return
+	}
+	defer configFile.Close()
+	log.Infof("creating default config file at %s", configFilePath)
+
+	if err := fetchAndWriteDefaultConfig(configFile); err != nil {
+		handleErrorAndFallback(err)
+		if err := os.Remove(configFilePath); err != nil {
+			log.Errorf("failed to remove config file at %s: %v", configFilePath, err)
+		}
+		return
+	}
+	log.Infof("config file created at %s with default configurations", configFilePath)
+}
+
+func isValidDirPath() bool {
+	if CustomConfigFilePath == constants.EmptyStr {
+		return false
+	}
+
+	info, err := os.Stat(CustomConfigFilePath)
+	if os.IsNotExist(err) || err != nil {
+		return false
+	}
+
+	if !info.IsDir() {
+		return false
+	}
+	return true
+}
+
+func areBothFlagsSet() bool {
+	return CustomConfigFilePath != constants.EmptyStr && ConfigFileLocation != filepath.Join(constants.DefaultConfigFilePath, constants.DefaultConfigName)
+}
+
+func setUpViperConfig(configFilePath string) {
+	viper.SetConfigName("dice")
```

Comment:
This is getting used by both `-o` and `-c` for `-c` arg we pass complete file name, either we should mandate prior in code that the expected config file name should always be `dice.toml` or avoid this setting for `-c` flag.

Reply:
Trying to keep things as simple as possible, for more robustness will extend the functionality 

In my opinion keeping file `dice.toml` would make more sense for even for the users to avoid confusion

Thoughts? @lucifercr07

Reply:
Will let others pitch in on this.

Reply:
Let's use a constant name for the config file for now.